### ISSUE TITLE
About the testing of novel views and novel projections

### ISSUE THREAD

**Issue by pipi-bibi** [yoterel/nepmap#4]
Hi, thanks for the great work. I am very interested in this repository and the paper. However, I could not find details in the repository concerning the division of the training and test sets, as well as the metrics calculated for the test set. Could you please provide the images and camera parameters corresponding to the 36 novel views and novel projections described in the paper?  Wish your reply!

**Comment by yoterel** [comment]
Hi,

> division of the training and test sets

I was slightly panicking initially because I didn't find this split anywhere in the data, but then I realized it was hard coded here:
https://github.com/yoterel/nepmap/blob/1c8557add2140275a3de8cbaa3d0c06040e7fdef/train.py#L323
Please use these for testing.
~Please manually exclude these indices from training (that is what I did), and use them for testing~ (edited the answer, see comment below!).
Note that indices start from 0.
Just as an example, if index 74 appears, you would just go to "0074.png" and that would be an image belonging to the test set.
To find the corresponding novel projection, you can use the file called "img2tex.json" which maps each image to a projection image in the projectors folder (the file can be found for each scene).

> metrics calculated for the test set

The paper shows the PSNR metric, which is computed using standard formula (see page 6 of paper, last paragraph):

$$10 \cdot log (\frac{max(Image)}{Loss_{img}})$$
where $Loss_{img}$ is the standard L2 loss over all the pixels (see equation (4) in the paper)

> Could you please provide the images and camera parameters corresponding to the 36 novel views

Yes, they are right there in the dataset folder, see the file called "transforms.json", which exists for every scene. (go to the entries corresponding to the image in the test set to find the camera info in blender format or opencv format). For example, if the index is 74, go to "0074.png" in transforms.json, and extract the camera info.

I apologize this isn't automated. If you by any chance write code to do this programmatically feel free to submit a pull request I would appreciate that !

**Comment by yoterel** [comment]
closing issue, please reopen if the issue persists.

**Comment by pipi-bibi** [comment]
Thank you. But I feel confused here:
_https://github.com/yoterel/nepmap/blob/1c8557add2140275a3de8cbaa3d0c06040e7fdef/train.py#L57_

I noticed that the images used for testing seem to be read from another folder named xxx_random. I guess maybe the test images are only read from such folders？Based on this situation, is it still necessary to manually exclude those frames used for testing from the training set?

**Comment by yoterel** [comment]
You are absolutely right that the xxx_random were used and not the original, which I forgot about.
Sorry for the confusion. You can forget what I said about excluding them from the training set (I edited the original answer for clarity).
